internal/util/verification.go

```diff
@@ -47,11 +48,25 @@ func AssertValid(verify types.Verification, data []byte) error {
 		hex.Encode(encodedSum, sum)
 		if string(encodedSum) != hash.Sum {
 			return ErrHashMismatch{
-				calculated: string(encodedSum),
-				expected:   hash.Sum,
+				Calculated: string(encodedSum),
+				Expected:   hash.Sum,
 			}
 		}
 	}
 
 	return nil
 }
+
+func GetHasher(verify types.Verification) (hash.Hash, error) {
+	hash := verify.Hash
+	if hash == nil {
```

Comment:
just use `verify.Hash == nil` and `verify.Hash.Function` below. Not used enough that a local var is helpful and you are masking the package named `hash` which can be confusing.